### PR TITLE
Control the solenoid from the Blower FSM.

### PR DESCRIPTION
**Pull request by jlebar**
<git-pr-chain>

#### Commits in this PR
1. Control the solenoid from the Blower FSM.
    
    Fixes https://github.com/RespiraWorks/VentilatorSoftware/issues/213.

#### [PR chain](https://github.com/jlebar/git-pr-chain)
1. 👉 #217 Control the solenoid from the Blower FSM. 👈 **YOU ARE HERE**
1. #218 Add NO_GUI_DEV_MODE, for running the controller without a GUI.
1. #205 Clarify comment on sensor scaling factor.
1. #219 Tune PID.

</git-pr-chain>

### CODE REVIEW

File: controller/lib/core/blower_pid.h [hunk 1]
```diff
@@ -23,11 +24,12 @@ limitations under the License.
...
 // Runs one step of the blower_pid module, which is responsible for spinning
-// the blower to attempt to achieve the given setpoint pressure.
+// the blower to attempt to achieve the given setpoint pressure and
+// opening/closing the expire solenoid.
 //
 // Puts readings from e.g. the pressure sensor into `readings`, and puts the
 // current fan power (range [0-1]) into `fan_power`.
-void blower_pid_execute(Pressure setpoint, SensorReadings *readings,
-                        float *fan_power);
+void blower_pid_execute(BlowerSystemState desired_state,
```
Comment (jkff) [inline controller/lib/core/blower_pid.h:32]: const&?

File: controller/lib/core/blower_pid.cpp [hunk 2]
```diff
@@ -49,11 +49,20 @@ void blower_pid_init() {
   myPID.SetMode(AUTOMATIC);
 }
 
-void blower_pid_execute(Pressure setpoint, SensorReadings *readings,
-                        float *fan_power) {
+void blower_pid_execute(BlowerSystemState desired_state,
```
Comment (jkff) [inline controller/lib/core/blower_pid.cpp:52]: const&?

File: controller/test/blower_fsm/blower_fsm_test.cpp [hunk 3]
```diff
@@ -16,23 +16,37 @@ limitations under the License.
...
 
+namespace {
+
 class BlowerFsmTest : public ::testing::Test {
 public:
   BlowerFsmTest() = default;
   void SetUp() override { blower_fsm_init(); }
 };
 
+// One could also accopmlish this with gmock matchers, but wow is this easier.
```
Comment (jkff) [inline controller/test/blower_fsm/blower_fsm_test.cpp:30]: accomplish

Comment (jkff) [inline controller/test/blower_fsm/blower_fsm_test.cpp:30]: Could also accomplish this with a comparator + overloaded ostream operator<< :) but this is ok too.

File: controller/test/blower_fsm/blower_fsm_test.cpp [hunk 4]
```diff
@@ -16,23 +16,37 @@ limitations under the License.
...
+  do {                                                                         \
+    BlowerSystemState s = (state);                                             \
+    EXPECT_FLOAT_EQ(s.pressure.cmH2O(), (pressure_cm_h2o));                    \
+    EXPECT_EQ(s.expire_valve_state, ValveState::valve_state);                  \
+  } while (0)
+
 TEST_F(BlowerFsmTest, InitiallyOff) {
   VentParams p = VentParams_init_zero;
-  EXPECT_EQ(blower_fsm_get_setpoint(p), kPa(0));
+  EXPECT_STATE_IS(blower_fsm_desired_state(p), 0, OPEN);
```
Comment (jkff) [inline controller/test/blower_fsm/blower_fsm_test.cpp:43]: Please annotate the "0" and other similar arguments below as `/*pressure_cm_h2o=*/` for readability

File: controller/lib/core/blower_pid.h (hunk 1 above)
Comment (jlebar) [reply controller/lib/core/blower_pid.h:32]: Thanks.

File: controller/test/blower_fsm/blower_fsm_test.cpp (hunk 3 above)
Comment (jlebar) [reply controller/test/blower_fsm/blower_fsm_test.cpp:30]: Done

File: controller/lib/core/blower_pid.cpp (hunk 2 above)
Comment (jlebar) [reply controller/lib/core/blower_pid.cpp:52]: Done.

**Pull request merged**